scripts/export-import-test.py

```diff
@@ -56,7 +56,206 @@ def run_command(cmd, cwd=None, capture_output=False):
+    os.environ["E2E_IMPORT_DB_DIR"] = export_path + os.sep
+    run_command("make test", cwd=test_worktree)
+    return 0
+
+
+def write_split_testfile(
+    export_dir, import_dir, case_name, header, export_lines, import_lines, db_dir
+):
+    export_path = os.path.join(export_dir, f"{case_name}.test")
+    import_path = os.path.join(import_dir, f"{case_name}.test")
+    os.makedirs(export_dir, exist_ok=True)
+    os.makedirs(import_dir, exist_ok=True)
+
+    def replace_placeholders(lines):
+        return [
+            line.replace("${KUZU_EXPORT_DB_DIRECTORY}", os.path.join(db_dir, ""))
+            for line in lines
+        ]
+
+    # Copying the dataset here is unnecessary and inefficient because we would
+    # have to remove it shortly after (import requires a fresh db instance).
+    def transform_import_header(header):
+        new_lines = []
+        for line in header.splitlines(keepends=True):
+            if line.startswith("-DATASET"):
+                parts = line.strip().split()
+                if len(parts) >= 2:
+                    parts[-1] = "EMPTY"
+                    line = " ".join(parts) + "\n"
+            new_lines.append(line)
+        return "".join(new_lines)
+
+    # Without this I run into an issue where the exported db seems to be
+    # deleted making the import fail. We still do import with the line
+    # -STATEMENT IMPORT DATABASE ...
+    def transform_import_lines(lines):
+        result = []
+        for line in lines:
+            if line.startswith("-IMPORT_DATABASE"):
+                continue
+            result.append(line)
+        return result
+
+    with open(export_path, "w") as f:
+        f.write(header.replace("${KUZU_EXPORT_DB_DIRECTORY}", os.path.join(db_dir, "")))
+        f.writelines(replace_placeholders(export_lines))
+
+    with open(import_path, "w") as f:
+        f.write(
+            transform_import_header(
+                header.replace("${KUZU_EXPORT_DB_DIRECTORY}", os.path.join(db_dir, ""))
+            )
+        )
+        f.writelines(transform_import_lines(replace_placeholders(import_lines)))
+
+
+def split_tests(root, output_dir, file, db_dir):
+    relative_path = os.path.relpath(file.name, root)
+    base_path = os.path.splitext(relative_path)[0]
+    export_dir = os.path.abspath(os.path.join(output_dir, "export", base_path))
+    import_dir = os.path.abspath(os.path.join(output_dir, "import", base_path))
+
+    header = ""
+    header_parsed = False
+    current_case = None
+    export_lines = []
+    import_lines = []
+    in_case = False
+    in_import = False
+    for line in file:
+        line = line.rstrip("\n")
+        if not header_parsed:
+            header += line + "\n"
+            if line.strip() == "--":
+                header_parsed = True
+            continue
+        if line.startswith("-CASE"):
+            # this is a spell to skip any cases that do not have a split
+            if in_case and in_import:
+                write_split_testfile(
```

Comment:
Alternatively could collect these in a vector and write them at once.

Reply:
We don't expect more than 10-20 cases on the higher end. Even upto a 1000 cases should be fine.

But current code is fine for now.